scripts/spack/packages/spheral/package.py

```diff
@@ -49,35 +49,32 @@ class Spheral(CachedCMakePackage, CudaPackage):
     depends_on('silo@4.10.2 +hdf5', type='build')
 
     # Zlib fix has been merged into conduit, using develop until next release.
-    depends_on('conduit@0.8.2 +shared +mpi +hdf5 -test ~parmetis', type='build', when='+mpi')
-    depends_on('conduit@0.8.2 +shared ~mpi +hdf5 -test ~parmetis', type='build', when='~mpi')
+    depends_on('conduit@0.8.2 +shared +mpi +hdf5~hdf5_compat -test ~parmetis', type='build', when='+mpi')
+    depends_on('conduit@0.8.2 +shared ~mpi +hdf5~hdf5_compat -test ~parmetis', type='build', when='~mpi')
+    depends_on('conduit@0.8.2 +shared +mpi +hdf5 -test ~parmetis', type='build', when='+mpi^hdf5@1.8.0:1.8')
+    depends_on('conduit@0.8.2 +shared ~mpi +hdf5 -test ~parmetis', type='build', when='~mpi^hdf5@1.8.0:1.8')
 
     depends_on('axom@0.7.0 ~shared +mpi +hdf5 -lua -examples -python -fortran -umpire -raja', type='build', when='+mpi')
     depends_on('axom@0.7.0 ~shared ~mpi +hdf5 -lua -examples -python -fortran -umpire -raja', type='build', when='~mpi')
 
     depends_on('caliper@2.8.0 ~shared ~adiak ~libdw ~papi ~libunwind +pic', type='build')
 
-    depends_on('opensubdiv@3.4.3', type='build')
-    depends_on('polytope@0.7 +python', type='build')
+    depends_on('opensubdiv', type='build')
+    depends_on('polytope +python', type='build')
 
     extends('python@3.9.10 +zlib +shared +ssl +tkinter', type='build')
 
     depends_on('py-numpy@1.23.4', type='build')
-    depends_on('py-numpy-stl@3.0.0', type='build')
-    depends_on('py-python-utils@2.4.0', type='build')
-    depends_on('py-matplotlib@3.3.4 backend=tkagg +fonts', type='build')
-    depends_on('py-pillow@9.2.0', type='build')
-    depends_on('py-decorator@5.1.1', type='build')
-    depends_on('py-h5py@3.7.0', type='build')
-    depends_on('py-docutils@0.19', type='build')
-    depends_on('py-cython@0.29.32', type='build')
-    depends_on('py-scipy@1.8.1', type='build')
-    depends_on('py-importlib-metadata@4.12.0', type='build')
-    depends_on('py-ats@exit', type='build')
-    depends_on('py-mpi4py@3.1.4', type='build', when='+mpi')
-
-    depends_on('py-sphinx@5.3.0', type='build')
-    depends_on('py-sphinx-rtd-theme@0.5.1', type='build')
+    depends_on('py-numpy-stl', type='build')
+    depends_on('py-matplotlib backend=tkagg +fonts', type='build')
+    depends_on('py-h5py', type='build')
+    depends_on('py-docutils', type='build')
+    depends_on('py-scipy', type='build')
+    depends_on('py-ats', type='build')
+    depends_on('py-mpi4py', type='build', when='+mpi')
+
+    depends_on('py-sphinx', type='build')
+    depends_on('py-sphinx-rtd-theme', type='build')
```

Comment:
We want to lock these down to working versions for reproducability across systems